terraform/provider/resource_teleport_bot.go

```diff
@@ -60,30 +60,15 @@ func GenSchemaBot(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
 				Type:        types.StringType,
 				Optional:    true,
 				Computed:    true,
-				Description: "The desired TTL for the token if one is created. If unset, a server default is used",
+				Description: "Deprecated. This field is not required anymore and has no effect.",
 			},
 			"token_id": {
 				Type: types.StringType,
-				// Implementation note: the Terraform provider differs from the
-				// actual API here. `token_id` is technically optional and when
-				// unset, a new random token is created automatically. This
-				// behavior doesn't play nicely with Terraform's resource
-				// management, so we'll instead make it required. Users can use
-				// a `random_password` resource with `teleport_provision_token`
-				// to achieve a similar result. See also:
-				// `example/bot.tf.example`
-				Required:    true,
+				// Implementation note: this is not used anymore, we can skip this
+				// This will go away eventually when we'll generate the bot provider instead
+				Optional:    true,
 				Sensitive:   true,
-				Description: "The bot joining token. If unset, a new random token is created and its name returned, otherwise a preexisting Bot token may be provided for IAM/OIDC joining.",
```

Comment:
Note: this description was wrong, it's not possible to unset the token in the TF provider.